### PR TITLE
Ieee802154 changes required for sub-ghz support

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

**Comment by tbursztyka** [comment]
hum, how can I make this PR dependent over https://github.com/zephyrproject-rtos/zephyr/pull/4094 so it won't show the commits of PR 4049?

File: include/net/ieee802154_radio.h [hunk 1]
```diff
@@ -30,6 +30,7 @@ enum ieee802154_hw_caps {
 	IEEE802154_HW_FILTER		= BIT(2),
```
Comment (jukkar) [inline include/net/ieee802154_radio.h]: Something does not sound right in commit message, could it be written something like this instead:

```
Sub-Ghz bands can have different limit of channels, typically 10 or more than thousand are possible.
This means that the device needs to expose such limit to the L2 which is unaware of frequency band logic. L2 will then allow user to select a proper channel.
```

**Review by jukkar — CHANGES_REQUESTED** [review]
Looks good, minor nit pick.

File: include/net/ieee802154_radio.h (hunk 1 above)
Comment (tbursztyka) [reply include/net/ieee802154_radio.h]: done

**Review by jukkar — DISMISSED** [review]

**Comment by pfalcon** [comment]
@tbursztyka :

> hum, how can I make this PR dependent over #4094 so it won't show the commits of PR 4049?

You can't. PRs posted should be independent from each other and not include unrelated changes. Locally, a developer can have "integration branch" where they merge separate PR branches for testing/further local development. Yeah, like those Linux subsystem maintainers.

That may be not too flexible, but promotes a workflow where a succession of small, easy to review PRs posted, instead of one big, long to review one (or so I think that it does ;-) ).

**Comment by pfalcon** [comment]
Commit: "net/ieee802154: Add lqi & rssi into net_pkt metadata structure":

Is it good model to store such data? For example, are there many users of such info which warrants spending memory on it for each and every packet? Secondly, RSSI/LQI apparently apply to a 15.4 packet, which for net_pkt becomes just a data fragment, and there can be many such data fragments with different RSSI/LQI's, how net_pkt's value get calculated? (Probably worth mention that in the commit message.)

**Review by pfalcon — CHANGES_REQUESTED** [review]
The commit message for "net/ieee802154: Add lqi & rssi into net_pkt metadata structure" has:

> It's simpler and make the values relevantly tighten to the received packet.

I guess "tighten" is a wrong word here, it should be "tied".

File: subsys/net/ip/l2/ieee802154/ieee802154_utils.h [hunk 2]
```diff
@@ -68,4 +68,19 @@ static inline void ieee802154_filter_pan_id(struct net_if *iface,
 	}
 }
 
+static inline bool ieee802154_verify_channel(struct device *dev, u16_t channel)
+{
+#ifdef CONFIG_NET_L2_IEEE802154_SUB_GHZ
+	const struct ieee802154_radio_api *radio = dev->driver_api;
+
+	if (radio->get_capabilities(dev) & IEEE802154_HW_SUB_GHZ) {
```
Comment (dleach02) [inline subsys/net/ip/l2/ieee802154/ieee802154_utils.h]: Why do we only verify SUB_GHZ channels and not the 2.4 stuff here?

File: subsys/net/ip/l2/ieee802154/Kconfig [hunk 3]
```diff
@@ -18,6 +18,10 @@ config NET_L2_IEEE802154_MGMT
 	select NET_MGMT_EVENT
 	default n
 
+config NET_L2_IEEE802154_SUB_GHZ
+	bool
+	default n
```
Comment (dleach02) [inline subsys/net/ip/l2/ieee802154/Kconfig]: This needs to have a "help" block to give the future user some guidance.

Comment (tbursztyka) [reply subsys/net/ip/l2/ieee802154/Kconfig]: done

**Comment by tbursztyka** [comment]
@pfalcon could you re-review? actually you reviewed patches that were not part of this PR.

**Comment by pfalcon** [comment]
@tbursztyka : Yep, my concern was a typo in a commit message which could lead to confusion. Dismissed my -1. Can't say much about specific technical 15.4 changes, leaving it to other folks.

**Comment by tbursztyka** [comment]
merged via CC1200 PR

**Pull request closed without merging**